File: php/insert_sort.php
<?php
	/**
	 * 插入排序
	 * @auther zhangzy
	 * @date 2014-02-17
	 */

	 function insert_sort($arr = array())
	 {
		$len = count($arr);
		$j = 0;
		for($i = 1; $i < $len; $i++){
			$j = $i - 1;
			$t = $arr[$i];
			while($j >= 0 && $arr[$j] > $t){
				$arr[$j+1] = $arr[$j];
				$j--;
			}
			$arr[$j+1] = $t;
		}
		return $arr;
	 }

	 $arr = array(2, 1, 5, 4,  3, 7, 6);
	 print_r(insert_sort($arr));

File: c/insert_sort.c
/*
 * 插入排序
 */
#include <stdio.h>
#define size 7
int main(void)
{
	void insert_sort(int *arr);
	int arr[size] = {2, 1, 5, 3, 7, 4, 1}, *p;
	insert_sort(arr);
	return 0;
}

void insert_sort(int *arr)
{
	int i, j, t;
	for(i = 1; i < size; i++){
		j = i - 1;
		t = arr[i];
		while(arr[j] > t && j >= 0){
			arr[j+1] = arr[j];
			j--;
		}
		arr[j+1] = t;
	}
	for(j = 0; j <= size; j++)
		printf("%3d", arr[j]);
	printf("\n");
}
